### DISCUSSION TITLE
Loading For Component Time load

### DISCUSSION THREAD

**Discussion by MahdiElahi** [dotnet/aspnetcore#55422] [Q&A]
i want use this [topbar.js](https://buunguyen.github.io/topbar/) 

but in blazor .net 8 how to show this topbar when user click on link forexample nav menu and how to hide after component load correctly 

how to do it?
i dont want write code for show and hide topbar in any component

how to config for all components to show/hide topbar ?

**Comment by maraf** [comment]
You can use `NativationManager.RegisterLocationChangingHandler` to show the loading state. Hiding it can be bound to `NavigationManager.LocationChanged` event, but individual components might be loading additional data and so hiding it when indivial component is done with loading data might be good idea.

I transfering the issue to the aspnetcore repository where Blazor folks can give more precise advice

**Comment by MahdiElahi** [comment]
@maraf 
please show a sample code 
i want config globally for all components

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. This doesn't seem to be a problem with the framework rather a question about how to use it.
We're converting this into a discussion so that our community members can step in and help you with solving it.
